src/runestone.ts

```diff
@@ -143,9 +145,10 @@ export class Runestone {
       fields,
       1,
       ([value]): Option<u32> =>
-        u128
-          .tryIntoU32(value)
-          .andThen((value) => (value < transaction.vout.length ? Some(value) : None))
+        u128.tryIntoU32(value).andThen((value) => {
+          if (etchPremine) return Some(value);
```

Comment:
This is incorrect, if value is greater than or equal to transaction.vout.length, this should always return None.